app/src/main/java/in/ac/bits_pilani/goa/ard/utils/Browser.java

```diff
@@ -0,0 +1,278 @@
+import static in.ac.bits_pilani.goa.ard.utils.BrowserUtil.ACTION_BUTTON_HEIGHT;
+import static in.ac.bits_pilani.goa.ard.utils.BrowserUtil.CLOSE_BUTTON_HEIGHT;
+import static in.ac.bits_pilani.goa.ard.utils.BrowserUtil.CLOSE_BUTTON_WIDTH;
+
+/**
+ * Browser opens web links.
+ * Includes default customizations which can be overridden.
+ *
+ * <h1>Usage</h1>
+ * <p>
+ * To open url with default customizations
+ * <pre>
+ * {@code
+ * new Browser(this).launchUrl("example.com");
+ * }
+ * </pre>
+ * </p>
+ * <p>
+ * To override values :
+ * <pre>
+ * {@code
+ * new Browser(this).setToolBarColor(ActivityCompat.getColor(R.color.cyan_500), this).launchUrl("example.com");
+ * }
+ * </pre>
+ * </p>
+ * @author Rushikesh Jogdand
+ */
+@SuppressWarnings("WeakerAccess")
+public final class Browser {
+
+    /**
+     * Host activity.
+     */
+    private final Activity hostActivity;
+
+    /**
+     * CustomTabsIntent Builder object.
+     */
+    private final CustomTabsIntent.Builder mBuilder;
+
+    /**
+     * CustomTabsIntent object.
+     */
+    private CustomTabsIntent mCustomTabsIntent;
+
+    /**
+     * Should url bar hide as the user scrolls down on the page.
+     */
+    private boolean urlBarHiding;
+
+    /**
+     * Whether to include share menu item.
+     */
+    private boolean shareInMenu;
+
+    /**
+     * Bundle ActionButton.
+     */
+    private Bundle actionButton;
+
+    /**
+     * Constructor initiates parameters to default values.
+     * @param hostActivity activity from which wab tab is launched.
+     */
+    public Browser(final Activity hostActivity) {
+        this.hostActivity = hostActivity;
+        mBuilder = new CustomTabsIntent.Builder();
+        initDefault();
+    }
+
+    /**
+     * Default customization.
+     */
+    private void initDefault() {
+        final int[] attrs = {R.attr.customTab_toolbar_color};
+        final TypedArray ta = hostActivity.obtainStyledAttributes(R.style.AppTheme, attrs);
+        setToolBarColor(ta.getColor(0, ActivityCompat.getColor(hostActivity, R.color.colorPrimary)));
+        ta.recycle();
+
+        setCloseButtonIcon((VectorDrawable) ContextCompat.getDrawable(hostActivity, R.drawable.ic_close_white_24dp));
+        setStartAnimation(R.anim.slide_in_up, R.anim.slide_out_down);
+        setExitAnimations(R.anim.slide_in_down, R.anim.slide_out_up);
+        setUrlBarHiding(true);
+        setShowTitle(true);
+        setShareInMenu(true);
+        disableActionButton();
+    }
+
+    /**
+     * Sets toolbar color.
+     * @param color The {@link Color}
+     * @return current instance.
+     */
+    public Browser setToolBarColor(@ColorInt final int color) {
+        mBuilder.setToolbarColor(color);
+        return this;
+    }
+
+    /**
+     * Sets the Close button icon for the custom tab.
+     * @param icon The icon {@link VectorDrawable}
+     * @return current instance.
+     */
+    public Browser setCloseButtonIcon(@NonNull final VectorDrawable icon) {
```

Comment:
I didn't know the method (`getBitmapFromVectorDrawable`) was applicable to any drawable. The code does look nice now!